### ISSUE TITLE
Paper Update

### ISSUE THREAD

**Issue by pGit1** [titu1994/Keras-Classification-Models#10]
https://arxiv.org/pdf/1802.01548.pdf - Supplants NASNet?

**Comment by pGit1** [comment]
Closing. This is just to keep you posted on papers.

**Comment by titu1994** [comment]
Thanks a ton ! I've read this paper, but to be honest, genetic algorithms are something I absolutely do not like. I have tons of bad experiences with them to make me give up on them completely.

I think the better paper is [Efficient NAS by Pham et al.](https://arxiv.org/abs/1802.03268), also by Barret Zoph and Quoc V. Le. Far more practical. 

And [Progressive NAS by Liu et al.](https://arxiv.org/abs/1712.00559) is also by Barret Zoph, and is far more tractable to work with. I was able to get some useable models for another course project by blindly following the paper in my P-NAS repo. To be sure, it is still highly time consuming, but considering I can let it run without supervision for a day and expect some pretty good results (not sota, but still close) is something I find practical.

Edit: Fixed up a few things, and added PNAS.

**Comment by pGit1** [comment]
This one is VERY interesting:
https://arxiv.org/pdf/1711.00489.pdf - Rather than Decay LR overtime Increase Batch Size since both have the effect of mitigating gradient noise. Brilliant idea actually...Wonder if a callback like this is implementable in keras. 

EDIT: Curious on your thoughts on how you would implement this. I could see this being implemented with for loops or something where we would keep call model.fit with different batch sizes at different points in the for loop.

**Comment by pGit1** [comment]
ping @titu1994

**Comment by titu1994** [comment]
That paper is nice, but not practical in my opinion. There is finite space for models in GPU memory, especially if you are not using multiple GPU. 

While the findings that learning rate and batchsize are both affecting performance are significant, it is simply too difficult to use such a technique for models that are large as NASNet, or deep like ResNets or memory hungry like DenseNets. 

I'm contrast, changing the learning rate is a fixed memory consumption technique, which is supported on all major frameworks. 

In any case, an implementation of this in Keras would be using the train_on_batch technique, or with .fit for 1 epoch within a for loop or something, and a change of the batch size for each call. I dunno if I recommend this method though.

**Comment by pGit1** [comment]
PERFECT answer!! Thank you!

**Comment by pGit1** [comment]
https://arxiv.org/pdf/1803.05407v1.pdf - An alternative to snapshot ensembles? Thoughts?

**Comment by pGit1** [comment]
ping @titu1994

**Comment by titu1994** [comment]
Sorry for the delay. I thought this was common practice in Tensorflow actually. I believe I read somewhere that they mentioned to keep the moving average of weights along with the weights are reach epoch since the moving average weights generally tend to boost the performance slightly. 

It's nice that it is now a paper to go along with that advice.

**Comment by titu1994** [comment]
Though, I must say, this is not my area of interest, so I just skimmed the paper. I may be horribly wrong.

**Comment by pGit1** [comment]
Is something like this possible to implement in Keras?
